spec/Section 3 -- Type System.md

```diff
@@ -1340,14 +1357,281 @@ Union type extensions have the potential to be invalid if incorrectly defined.
+
+* No entry should be added to the unordered map for any other members defined
+  on the Tagged type.
+
+Following are examples of input coercion for a Tagged type with a
+`String` member `a` and a required (non-null) `Int!` member `b`:
+
+```graphql example
+tagged ExampleInputTagged {
+  a: String
+  b: Int!
+}
+```
+
+Literal Value            | Variables               | Coerced Value
+------------------------ | ----------------------- | ---------------------------
+`{ a: "abc", b: 123 }`   | `{}`                    | Error: Exactly one key must be specified
+`{ a: null, b: 123 }`    | `{}`                    | Error: Exactly one key must be specified
+`{ b: 123 }`             | `{}`                    | `{ b: 123 }`
+`{ a: $var, b: 123 }`    | `{ var: null }`         | Error: Exactly one key must be specified
+`{ a: $var, b: 123 }`    | `{}`                    | `{ b: 123 }`
+`{ b: $var }`            | `{ var: 123 }`          | `{ b: 123 }`
+`$var`                   | `{ var: { b: 123 } }`   | `{ b: 123 }`
+`"abc123"`               | `{}`                    | Error: Incorrect value
+`$var`                   | `{ var: "abc123" } }`   | Error: Incorrect value
+`{ a: "abc", b: "123" }` | `{}`                    | Error: Exactly one key must be specified
+`{ b: "123" }`           | `{}`                    | Error: Incorrect value for member {b}
+`{ a: "abc" }`           | `{}`                    | `{ a: "abc" }`
+`{ b: $var }`            | `{}`                    | Error: No keys were specified
+`$var`                   | `{ var: { a: "abc" } }` | `{ a: "abc" }`
+`{ a: "abc", b: null }`  | `{}`                    | Error: Exactly one key must be specified
+`{ b: $var }`            | `{ var: null }`         | Error: {b} must be non-null.
+`{ b: 123, c: "xyz" }`   | `{}`                    | Error: Unexpected member {c}
+
+
+
+**Type Validation**
+
+Tagged types have the potential to be invalid if incorrectly defined. This set
+of rules must be adhered to by every Tagged type in a GraphQL schema.
+
+1. A Tagged type must define one or more members.
+2. For each member of a Tagged type:
+   1. The member must have a unique name within that Tagged type;
+      no two members may share the same name.
+   2. The member must not have a name which begins with the
+      characters {"__"} (two underscores).
+3. {IsOutputType(taggedType)} and {IsInputType(taggedType)} cannot both be {false}.
+
+### Member Deprecation
+
+Members in a Tagged type may be marked as deprecated as deemed necessary by
+the application. It is still legal to query for these members on outputs or
+supply these members in inputs (to ensure existing clients are not broken by
+the change), but the members should be appropriately treated in documentation
+and tooling.
+
+When using the type system definition language, `@deprecated` directives are
+used to indicate that a field is deprecated:
+
+```graphql example
+tagged ExampleType {
+  oldField: String @deprecated
```

Comment:
thought: the `@deprecated` directive needs its locations updated to allow on tagged members

Reply:
Done :+1: